File: src/scraper/entities/weapon_level_rank.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, ManyToMany, JoinTable, Unique } from 'typeorm';
import { WeaponEntity } from './weapon.entity';
import { ItemEntity } from './item.entity';
import { SkillEntity } from './skill.entity';

@Entity()
@Unique(['level', 'rank'])
export class WeaponLevelRank {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  level: number;

  @Column()
  rank: number;

  @Column()
  stat1_name: string;

  @Column()
  stat1_value: number;

  @Column()
  stat2_name: string;

  @Column()
  stat2_value: number;

  @Column()
  ascension_max_level: number

  @Column()
  about: string

  @ManyToOne(() => WeaponEntity, weapon => weapon.levelRanks)
  weapon: WeaponEntity;

  @ManyToMany(() => ItemEntity)
  @JoinTable()
  items: ItemEntity[];

  @Column('int', { array: true })
  itemCounts: number[];

  @ManyToMany(() => SkillEntity)
  @JoinTable()
  skills: SkillEntity[];
}

File: mobile/lib/domains/weapon/weapon_level_rank_entity.dart

import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:mobile/domains/item/item_entity.dart';
import 'package:mobile/domains/weapon/weapon_skill_entity.dart';

part 'weapon_level_rank_entity.freezed.dart';
part 'weapon_level_rank_entity.g.dart';
@freezed
class WeaponLevelRankEntity with _$WeaponLevelRankEntity {
  const factory WeaponLevelRankEntity({
    int? id,
    int? level,
    int? rank,
    @JsonKey(name: 'stat1_name')String? stat1Name,
    @JsonKey(name: 'stat1_value')String? stat1Value,
    @JsonKey(name: 'stat2_name')String? stat2Name,
    @JsonKey(name: 'stat2_value')String? stat2Value,

    @JsonKey(name: 'ascension_max_level')int? ascensionMaxLevel,
    String? about,
    List<int>? itemCounts,
    List<ItemEntity>? items,
    List<WeaponSkillEntity>? skills,
  }) = _WeaponLevelRankEntity;

  factory WeaponLevelRankEntity.fromJson(Map<String, dynamic> json) =>
      _$WeaponLevelRankEntityFromJson(json);
}